### ISSUE TITLE
[MSHARED-800] Remove Maven version from pom.properties

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archiver#221]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MSHARED-800](https://issues.apache.org/jira/browse/MSHARED-800?redirect=false)** and commented

Currently, generated `META-INF/maven/${groupId}/${artifactId}/pom.properties` contains `#Created by Apache Maven ${maven.version}`, which is not a reproducible value

we should either remove the version, either use the same value as the `Created-By` manifest entry as defined in MSHARED-799

---

**Affects:** maven-archiver-3.3.0

**Issue Links:**
- [MSHARED-513](https://issues.apache.org/jira/browse/MSHARED-513) Regression: Created-By row in manifest does not include Maven version
 (_**"is caused by"**_)
- [MSHARED-799](https://issues.apache.org/jira/browse/MSHARED-799) Change "Created-By" manifest entry value to be reproducible

- [MSHARED-833](https://issues.apache.org/jira/browse/MSHARED-833) make pom.properties entries order reproducible

**Remote Links:**
- [GitHub Pull Request #5
](https://github.com/apache/maven-archiver/pull/5)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Insane [Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), did you read my mind? I had the same ticket idea a couple of days ago ;)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I think we should leave `# Created by Apache Maven` only at most. Everything else is duplicate information since it can be provided via `MANIFEST.MF`. Though, I dislike the `Created by` value because it contradicts the base value in `MANIFEST.MF`.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

yes, I'm trying to reproduce every release on which I vote and track precisely every little detail, with precise JIRA issue: it's good to know you've got the same idea :)

yes, we can leave the value as simple as `# Created by Apache Maven`, that will be sufficient

one question I had while looking at code: should we explicitely sort entries also? I did not get any issue, but I fear the current order is not really kept by design but by chance only...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The order isn't stable because, as far as I know, it backed by a hash. It should be at most a linked hash map or a lnked tree set. At least something stable. Let's have another ticket for that.

```java

package java.util.jar;

import java.io.FilterInputStream;
import java.io.DataOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.io.IOException;
import java.util.Map;
import java.util.HashMap;
import java.util.Iterator;

/**
 * The Manifest class is used to maintain Manifest entry names and their
 * associated Attributes. There are main Manifest Attributes as well as
 * per-entry Attributes. For information on the Manifest format, please
 * see the
 * <a href="../../../../technotes/guides/jar/jar.html">
 * Manifest format specification</a>.
 *
 * @author  David Connelly
 * @see     Attributes
 * @since   1.2
 */
public class Manifest implements Cloneable {
    // manifest main attributes
    private Attributes attr = new Attributes();

    // manifest entries
    private Map<String, Attributes> entries = new HashMap<>();
 
```

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

I was not talking about `MANIFEST.MF` but `pom.properties`

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Sorry, my bad, but the fact remains. It is backed by a `Hastable`.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), branch created. Have a look, but I think that even this line is just waste. Consider that someone provided a custom manifest file. We'd still would write the comment.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

As for the sort, we could leverage this: https://stackoverflow.com/a/25059741/696632 while groupId, artifactId and version come always as first element.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), please have a look at the branch, I'd like to finalize this release.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) +1, go!

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [44966b9487667adb4bd061c1549fcd9f27ea0522](https://gitbox.apache.org/repos/asf?p=maven-archiver.git;a=commit;h=44966b9487667adb4bd061c1549fcd9f27ea0522).
